.github/workflows/ci.yml

```diff
@@ -292,6 +292,103 @@ jobs:
             dist-newstyle
           key: ${{ env.CABAL_CACHE_VERSION }}-cabal-${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles(format('cabal.GHC-{0}.config', matrix.ghc)) }}-${{ github.sha }}
 
+  build-docs:
+    name: "Build and deploy documentation to GitHub pages"
+    needs: [coverage]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+    steps:
+      - name: Check out repository (no submodules)
+        uses: actions/checkout@v4
+
+      - name: Cache Python environment
+        uses: actions/cache@v4
+        with:
+          path: doc/.venv
+          key: ${{ matrix.os }}-venv-${{ hashFiles('doc/scripts/requirements.txt') }}
+
+      - name: Build HTML documentation
+        run: make -C doc html
+
+      - name: Gather HPC coverage files and copy to HTML docs
+        run: |
+          .github/ci.sh collect_all_html
+          mkdir -p doc/_build/html/coverage
+          cp -r all-html/. doc/_build/html/coverage
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      # Adapted from:
+      # https://github.com/brechtm/rinohtype/commit/8a23be41d39d748955b139f0fabefe3205a4fd93
+      # Based on the Stack Overflow thread:
+      # https://stackoverflow.com/questions/72089650/how-to-host-multiple-version-of-a-sphinx-based-documentation-on-github
+      - name: Determine documentation publishing directory
+        id: doc-publish-dir
+        run: |
+          python3 -c "
+          import os, re
+          github_ref = os.environ.get('GITHUB_REF')
+          if github_ref == 'refs/heads/master':
+            target = 'master'
+          elif github_ref.startswith('refs/tags/'):
+            # Matches vx.y or vx.y.z
+            m = re.match(r'^refs/tags/v([0-9]+\.[0-9]+(?:\.[0-9]+)?)$', github_ref)
+            target = m.group(1)
+          else:
+            target = 'test-version'
+          print(f'echo \"target={target}\" >> $GITHUB_OUTPUT')"
+
+      # Using peaceiris/actions-gh-pages, as it provides more control over the
+      # deployment.
```

Comment:
More control than what? [`actions/upload-pages-artifact`](https://github.com/actions/upload-pages-artifact)?

Reply:
Clarified in d7e7da16e51494effa6e083db0eb175086926adc